From 7ca66c10c05168c7d342df7c7a70d4a1ae0629f7 Mon Sep 17 00:00:00 2001
From: Eric Nelson <eric.nelson@boundarydevices.com>
Date: Sat, 28 Jun 2014 09:45:09 -0700
Subject: [PATCH] vpu_wrapper: fix tests of return value from IOGetVirtMem

IOGetVirtMem() returns a pointer or specifically -1 (MAP_FAILED)
in the case of failure.

Upstream-Status: Pending

Signed-off-by: Laci Tele <laci@boundarydevices.com>
Signed-off-by: Eric Nelson <eric.nelson@boundarydevices.com>
---
 vpu_wrapper.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vpu_wrapper.c b/vpu_wrapper.c
index 9249174..148c5df 100755
--- a/vpu_wrapper.c
+++ b/vpu_wrapper.c
@@ -6653,7 +6653,7 @@ VpuDecRetCode VPU_DecGetMem(VpuMemDesc* pInOutMem)
 		return VPU_DEC_RET_FAILURE;
 	}
 	ret=IOGetVirtMem(&buff);
-	if(ret==0) //if(ret!=RETCODE_SUCCESS)
+	if(ret == -1) //if(ret==MAP_FAILED)
 	{
 		VPU_ERROR("%s: get virtual memory failure: size=%d, ret=%d \r\n",__FUNCTION__,buff.size,ret);
 		return VPU_DEC_RET_FAILURE;
@@ -8277,7 +8277,7 @@ VpuEncRetCode VPU_EncGetMem(VpuMemDesc* pInOutMem)
 		return VPU_ENC_RET_FAILURE;
 	}
 	ret=IOGetVirtMem(&buff);
-	if(ret==0) //if(ret!=RETCODE_SUCCESS)
+	if(ret == -1) //if(ret==MAP_FAILED)
 	{
 		VPU_ENC_ERROR("%s: get virtual memory failure: size=%d, ret=%d \r\n",__FUNCTION__,buff.size,(UINT32)ret);
 		return VPU_ENC_RET_FAILURE;
-- 
1.9.1

